android-components/components/service/firefox-accounts/src/main/java/mozilla/components/service/fxa/manager/FxaAccountManager.kt

```diff
@@ -610,19 +595,16 @@ open class FxaAccountManager(
                     }
                 }
                 val finalize = suspend {
-                    withRetries(logger, MAX_NETWORK_RETRIES) { finalizeDevice(via.authData.authType) }
+                    val capabilities = deviceConfig.capabilities.map { it.into() }.toSet()
+                    withRetries(logger, MAX_NETWORK_RETRIES) {
+                        account.initializeDevice(deviceConfig.name, deviceConfig.type.into(), capabilities)
+                    }
                 }
                 // If we can't 'complete', we won't run 'finalize' due to short-circuiting.
                 if (completeAuth() is Result.Failure || finalize() is Result.Failure) {
-                    resetAccount()
                     Event.Progress.FailedToCompleteAuth
                 } else {
-                    via.authData.declinedEngines?.let { persistDeclinedEngines(it) }
-                    if (authenticationSideEffects("CompletingAuthentication:AuthData")) {
-                        Event.Progress.CompletedAuthentication(via.authData.authType)
-                    } else {
-                        Event.Progress.FailedToCompleteAuth
-                    }
+                    Event.Progress.CompletedAuthentication(via.authData.authType)
```

Comment:
That's actually a pretty big change. It used to be a transitional state causing a side effect that might change the flow of transitional states (`completingAuth` -> `authError`), but as we are moving inner flow into the A-S, it will become a stable state that would trigger side effects that might relaunch the state machine (`completedAuth` -> `authError`). Looks like the way to go!